src/workflows/recipes/components/FilterObjectForm.js

```diff
@@ -656,6 +676,17 @@ export class SamplingInput extends React.PureComponent {
   renderStableInputs = () => {
     const { disabled, value, formErrors } = this.props;
 
+    let step = 1;
+    // If the current value rate has a decimal (e.g. 0.1% or 12.3%) then
+    // make the step 0.1.
```

Comment:
I'm not really a fan of this whole step thing. It seems pretty complicated. Do you think it is a useful addition to the UI?

Reply:
I had that feeling too. But I found it pretty neat. Now the little up and down errors "match" the input. If it was "13" and press down it goes to "12". And if it was "0.3" and press down it goes to "0.2". 
Despite its complexity it's pretty self-contained. Did you test it manually? I think it's worth keeping but with hindsight it might be a bit overkill to have made.

Reply:
If it seems useful, lets keep it (possibly with some more robust tests). I don't usually find these step buttons useful in general, but it's possible I'm an outlier.